code/modules/mob/living/carbon/human/species_types/plasmamen.dm

```diff
@@ -61,6 +61,8 @@
 /datum/species/plasmaman/on_species_gain(mob/living/carbon/C, datum/species/old_species, pref_load)
 	. = ..()
 	C.set_safe_hunger_level()
+	if(!(old_species.inherent_biotypes & MOB_MINERAL)) // if the mob was previously not of the plasmaman biotype
```

Comment:
You should put this on the /datum/species type, otherwise this will keep happening with different species

Reply:
You don’t need to compare the biotypes between the two species. MOB_MINERAL is immune to most toxins, so if at any point any species becomes MOB_MINERAL toxins should be cleared. 

You’re right in that we can’t track where stuff came from, but tbh species changes to mineral are rare enough, that I think it’s fine to just full-clear the toxin damage in every scenario (when changing to MOB_MINERAL) rather than have some people get incurably sick. No one is going to jump in a plasmariver anytime soon to heal their toxin damage

Reply:
It's not great either way, at least this way people will get toxin locked the least